File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOV3NoiseParamModel.kt
//
//  IDOV3NoiseParamModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///
/// Set Environmental Noise Volume On/Off and Threshold Event

open class IDOV3NoiseParamModel(
    mode: Int,
    startHour: Int,
    startMinute: Int,
    endHour: Int,
    endMinute: Int,
    highNoiseOnOff: Int,
    highNoiseValue: Int
) : IDOBaseModel {

    /// All-day environmental noise volume switch
    /// 1: On
    /// 0: Off 
    @SerializedName("mode")
    var mode: Int = mode

    /// Start time (hour)
    @SerializedName("start_hour")
    var startHour: Int = startHour

    /// Start time (minute)
    @SerializedName("start_minute")
    var startMinute: Int = startMinute

    /// End time (hour)
    @SerializedName("end_hour")
    var endHour: Int = endHour

    /// End time (minute)
    @SerializedName("end_minute")
    var endMinute: Int = endMinute

    /// Threshold switch
    /// 1: On
    /// 0: Off                      
    @SerializedName("high_noise_on_off")
    var highNoiseOnOff: Int = highNoiseOnOff

    /// Threshold value
    @SerializedName("high_noise_value")
    var highNoiseValue: Int = highNoiseValue


    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOV3NoiseParamModel.swift
//
//  IDOV3NoiseParamModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Set Environmental Noise Volume On/Off and Threshold Event
@objcMembers
public class IDOV3NoiseParamModel: NSObject, IDOBaseModel {
    /// All-day environmental noise volume switch
    /// 1: On
    /// 0: Off 
    public var mode: Int
    /// Start time (hour)                                            
    public var startHour: Int
    /// Start time (minute)                                          
    public var startMinute: Int
    /// End time (hour)                                              
    public var endHour: Int
    /// End time (minute)                                            
    public var endMinute: Int
    /// Threshold switch
    /// 1: On
    /// 0: Off                      
    public var highNoiseOnOff: Int
    /// Threshold value                                              
    public var highNoiseValue: Int
    
    enum CodingKeys: String, CodingKey {
        case mode = "mode"
        case startHour = "start_hour"
        case startMinute = "start_minute"
        case endHour = "end_hour"
        case endMinute = "end_minute"
        case highNoiseOnOff = "high_noise_on_off"
        case highNoiseValue = "high_noise_value"
    }
    
    public init(mode: Int,startHour: Int,startMinute: Int,endHour: Int,endMinute: Int,highNoiseOnOff: Int,highNoiseValue: Int) {
        self.mode = mode
        self.startHour = startHour
        self.startMinute = startMinute
        self.endHour = endHour
        self.endMinute = endMinute
        self.highNoiseOnOff = highNoiseOnOff
        self.highNoiseValue = highNoiseValue
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

